### ISSUE TITLE
Window still visible on macOS after credential access confirmation

### ISSUE THREAD

**Issue by mpas97** [keepassxreboot/keepassxc#4904]
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )
When being in the browser, for instance, and credentials are requested, the access confirmation dialog is shown. So far so good. But as soon as I tap "allow" keepass will create a window, if it's not already existing. In my case I have a tray icon with hidden window normally. So after having confirmed, I have to re-hide the keepass window first again. That's just annoying.

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. Unlock db, hide window
2. open website you want to log in
3. allow/deny credential request
4. there you have again a window to hide

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
"allow", "deny" should only do their actions without producing new windows I don't want nor need.

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
Tapping "allow", "deny" are producing an open window, even if it was hidden before.

## Context
[NOTE]: # ( Give us any additional information you may have. )

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - 2.6.0-beta1

[NOTE]: # ( Pick choices based on your environment )
Operating System: macOS

**Comment by droidmonkey** [comment]
Oops macos bug, easy fix though

**Comment by mpas97** [comment]
@varjolintu I also experience the issue after autotype confirmation dialogs. Will that be fixed by #4909 too? Anyway, thank you for fixing it! :)

**Comment by varjolintu** [comment]
@mpas97 I'll check that too.

**Comment by mpas97** [comment]
@droidmonkey When is a new pre-release coming? In the last month a lot of fixes and additions where made. As now also the the fix for that issue has been included, I would be very happy to see one soon.

**Comment by droidmonkey** [comment]
We are doing a final release today

**Comment by mpas97** [comment]
This issue still persists but it happens not as often as back then (where it happened always). Shall I reopen the issue or create a new one? But I'm not able to tell you in which cases this still happens... It's like on random occassion, but I would say that the desired hidding occurs less frequent than the unwanted persistent window.

**Comment by droidmonkey** [comment]
We can't fix random, unfortunately

**Comment by mpas97** [comment]
As this really annoys me every day, I retested all different workflows to find any pattern. Now, I think to have found some pattern as it was reproducible a few times in a row.
Hiding the window through the yellow minus in the top left corner seems to do the job correctly. The desired window state becomes hidden, so it again disappears automatically with the app icon together. When I though hide the window through Cmd+M or through the "tray icon"->"Hide window", the main app icon persits after password fill requests.
Statistically this matches my last comment where the right things happen less often than the wrong ones. As only 1 of 3 buttons seem to change the desired window/icon state.

**Comment by mpas97** [comment]
When KeePass is started and unlocked, the window gets hidden automatically, but the icon persists too. I then have to click on the icon first, so the window gets visible. Then, as written above, clicking the "yellow minus" solves the problem. Any other hidding option doesn't change the future hiding behaviour.

**Comment by droidmonkey** [comment]
So you have minimize on unlock enabled?

**Comment by mpas97** [comment]
Yes, I have.

**Comment by droidmonkey** [comment]
If you don't want this behavior: "unlocked, the window gets hidden automatically, but the icon persists too. I then have to click on the icon first, so the window gets visible." Then disable that setting. I don't think there is an actual problem here, your settings might not be setup correctly.

**Comment by mpas97** [comment]
Of course, I could deactivate that. But at the end I just want to have KeePass always running, with tray icon only.  So that the window minimizes automatically, is what I acually want. But that the main icon and window is not disappearing as expected (as I have the "hide window when minimized" active too) is the same discussed issue. That's the whole topic here. I just wrote it, so you can make sure when fixing the issue, that the hiding also works as intended when minimizing occurs automatically. Or in other words:

minimize through:
- "yellow minus" button => works
- Cmd+M => **not** working
- "tray icon"->"Hide window" => **not** working
- "minimize on unlock" setting => **not** working

**Comment by droidmonkey** [comment]
That is much clearer thank you

**Comment by droidmonkey** [comment]
I could not replicate the problem on Catalina. All 4 scenarios you put above cause the window to hide completely to the tray.

* Show system tray icon -> **ON**
* Hide window to system tray when minimized -> **ON**

**Comment by mpas97** [comment]
I just reset all settings and reinstalled the application. I then only reactivated the settings mentioned above, but the issue still persists. Probably important to (re-)mention (and that's probably why you couldn't reproduce it), is that the icon disappears in all 4 cases just after this minimizing operation as expected. Until here everything works. But then, when eg. the dialogs for password requests are poping up and we click "allow" (see main issue description) the window/icon is only rehidden correctly if it was hidden by "minus" previously. Did you check that?

**Comment by mpas97** [comment]
Here a video. A few times you see it working after a "minus" hide (second 8, for instance) and at time 20s you see the icon not disappearing after a tray icon close and on second 38 a popup after a cmd+m hiding. So again, the hiding method only impacts the future automatic hiding operations as after request dialogs.


[macIconBug.zip](https://github.com/keepassxreboot/keepassxc/files/5437920/macIconBug.zip)

**Comment by mpas97** [comment]
@droidmonkey did you get my last comments? Just to know if it is still not reproducible for you. As for me it is, even after reinstalling and clearing all settings.

**Comment by droidmonkey** [comment]
Watched your video, that is an extremely specific situation that this occurs. Does it happen when you are not clicking so fast? This could just be a timing issue.

**Comment by droidmonkey** [comment]
Yes I can reproduce the problem easily now:

1. Hide KeePassXC to tray using the tray icon "toggle"
2. Go to website that requests allow/deny credentials
3. Allow credentials, KeePassXC properly minimizes but does not hide from dock

**Comment by mpas97** [comment]
@droidmonkey to keep the issue updated, the "not reproducible" label is no longer true (see your last comment) and should be removed here.

**Comment by droidmonkey** [comment]
No longer an issue in 2.7.0
